### ISSUE TITLE
org.jetbrains.kotlin:kotlin-test:1.7.10 dependency error

### ISSUE THREAD

**Issue by darky** [badoo/Reaktive#675]
```
> Could not resolve all files for configuration ':testRuntimeClasspath'.
   > Could not resolve org.jetbrains.kotlin:kotlin-test-junit5:1.7.10.
     Required by:
         project : > org.jetbrains.kotlin:kotlin-test:1.7.10
      > Module 'org.jetbrains.kotlin:kotlin-test-junit5' has been rejected:
           Cannot select module with conflict on capability 'org.jetbrains.kotlin:kotlin-test-framework-impl:1.7.10' also provided by [org.jetbrains.kotlin:kotlin-test-junit:1.6.21(junitRuntime)]
   > Could not resolve org.jetbrains.kotlin:kotlin-test-junit:1.6.21.
     Required by:
         project : > com.badoo.reaktive:reaktive-testing:1.2.2 > com.badoo.reaktive:reaktive-testing-jvm:1.2.2
      > Module 'org.jetbrains.kotlin:kotlin-test-junit' has been rejected:
           Cannot select module with conflict on capability 'org.jetbrains.kotlin:kotlin-test-framework-impl:1.6.21' also provided by [org.jetbrains.kotlin:kotlin-test-junit5:1.7.10(junit5Runtime)]
```

Maybe this PR https://github.com/badoo/Reaktive/pull/666 will fix :point_up: ?

**Comment by arkivanov** [comment]
Thanks for the report! What Kotlin and Reaktive versions are you using?

**Comment by darky** [comment]
Reaktive: 1.2.2
Kotlin: 1.7.10

**Comment by arkivanov** [comment]
Thanks! If I understand it correctly, the issue is a bit different. It looks like `kotlin-test-junit` conflicts with `kotlin-test-junit5`. The former is used by `reaktive-testing`, the latter perhaps used in your project.

You can try using `kotlin-test-junit` instead if that's an option. Otherwise it should be possible to exclude it as follows:

```
implementation("com.badoo.reaktive:reaktive-testing:1.2.2") {
    exclude("org.jetbrains.kotlin", "kotlin-test-junit")
}
```

I have found similar issue: https://youtrack.jetbrains.com/issue/KT-46090

@CherryPerry what do you think about this issue?

**Comment by CherryPerry** [comment]
@arkivanov Looks like you are right, I do not think we can solve the issue other than ditching `kotlin-test` from transitive dependencies. We do not use any advanced functionality, so we can implement all assertion methods manually by throwing `AssertionError` like [RxJava](https://github.com/ReactiveX/RxJava/blob/5bb119b148/src/main/java/io/reactivex/rxjava3/observers/BaseTestConsumer.java#L350) does. Gradle should be fine, but not sure about JS tests, better to check in the corresponding PR.

**Comment by arkivanov** [comment]
Thanks @CherryPerry ! I will try to get rid of `kotlin-test-junit`. This fix would be a good fit for `v1.3.0`.
